### DISCUSSION TITLE
Emporia Vue 2 won't update to 2023.9.1

### DISCUSSION THREAD

**Discussion by rjing001** [emporia-vue-local/esphome#219] [Q&A] [answered]
Attempting to update from 2023.8.3 to 2023.9.1 in esphome.
Getting the following error message:

```
INFO ESPHome 2023.9.1
INFO Reading configuration /config/esphome/emporiavue2.yaml...
INFO Cloning https://github.com/flaviut/esphome.git@emporia-vue-2022.4.0
Failed config

external_components: [source /config/esphome/emporiavue2.yaml:5]
  - 
    Remote branch emporia-vue-2022.4.0 not found in upstream origin.
    source: github://flaviut/esphome@emporia-vue-2022.4.0
    components: 
      - emporia_vue
```

I've had the emporia vue 2 now for over a year and have done many updates without issue.
Also, I have several other esphome devises that took the update without a problem.

Can anyone offer assistance?

**Comment by flaviut** [comment]
Update the repo URL:   `  - source: github://emporia-vue-local/esphome@dev`

**Reply by rjing001** [reply]
Sorry not well versed in all of this, could you be more specific.  Update it to what?

**Reply by flaviut** [reply, marked as answer]
weird, something censored that. Update the source line to `  - source: github://emporia-vue-local/esphome@dev` in your config

**Reply by rjing001** [reply]
Ok that worked.  The only thing I don't understand is, when this changed.  It was working fine for so long.  And, how was I suppose to know that this changed, apart from visiting github everyday.  I'm not being critical here, you've done a fantastic job with this.  I'm just not as well versed in the processes as some others may be.  I'm trying to understand what the expectation is for the users here.

**Reply by clowrey** [reply]
Don't update unless you need to change your ESPhome config as a general practice. Beyond that expect it to break things and you'll have to investigate as you have done here. This is true for all ESPhome devices.

**Reply by flaviut** [reply]
I definitely don't think you're being too critical. This is a frustrating experience.

I also generally avoid updating until I have a need for it and am prepared to spend a little bit of time fixing things.

In this particular case, you can also troubleshoot by noticing the error message seems to be around the `external_components` area, and checking back in the sample config to see what that should look like. Or just ask here, where I'm (and other folks) are happy to help :)

**Reply by Sc00bs110** [reply]
Hi Flaviut, Thank for all the great work on this, I have the same problem with the Github having changed  and am getting the error:

Failed config

sensor.emporia_vue: [source /config/esphome/emporia.yaml:67]
  
  Platform not found: 'sensor.emporia_vue'.  

My Emporia is offline in EspHome and as a result I am unable to update the Github in the code. 

Is there anyway to update other than connecting the Emporia up to my pc? 

Thanks 

JohnB

**Reply by flaviut** [reply]
> Platform not found: 'sensor.emporia_vue'.

Does the start of your config match the example config?

```
external_components:
  - source: github://emporia-vue-local/esphome@dev
    components:
      - emporia_vue
```

if it does, try [clicking "clear build files"](https://esphome.io/guides/getting_started_hassio.html) and trying again

> My Emporia is offline in EspHome and as a result I am unable to update the Github in the code.
>
> Is there anyway to update other than connecting the Emporia up to my pc?

Unfortunately, no. This issue is unrelated to your other issue, but if the dashboard can't connect to it remotely, then you need to plug it in to update.

**Reply by Sc00bs110** [reply]
Thanks @flaviut

It was pointing to the old Github page in the ESPhome Yaml code and was coming up with an error because of that. 
I changed the Yaml code inESPHome to the the new Github page but am unable to update the Emporia as it is offline (even though I am still getting readings from it in Home Assistant).

I asked it to clean the build files but now get the error: 
"INFO ESPHome 2023.11.4
INFO Reading configuration /config/esphome/emporia.yaml...
Failed config

external_components: [source /config/esphome/emporia.yaml:5]
  - 
    Source is not a file system path, in expected github://username/name[@branch-or-tag] or github://pr#1234 format!.
    source: https://github.com/emporia-vue-local/esphome
    components: 
      - emporia_vue"

If I go to the log for the Emporia I get the message:

INFO ESPHome 2023.11.4
INFO Reading configuration /config/esphome/emporia.yaml...
Failed config

external_components: [source /config/esphome/emporia.yaml:5]
  - 
    Source is not a file system path, in expected github://username/name[@branch-or-tag] or github://pr#1234 format!.
    source: https://github.com/emporia-vue-local/esphome
    components: 
      - emporia_vue

Not sure if you have any other suggestions as to how I might be able to fix it?

**Reply by Sc00bs110** [reply]
All working now @flaviut , thank you for your help :-)

**Comment by jklap** [comment]
> Don't update unless you need to change your ESPhome config as a general practice. Beyond that expect it to break things and you'll have to investigate as you have done here. This is true for all ESPhome devices.

Or update on a regular basis so when it does break you have less you have to dig through to work out the problem. Plus you are then getting fixes, improvements, etc. Given this is home automation, I prefer to stay current.

I've updated (at least) every couple of releases for over a year and have run into few issues, usually related to things like config/yaml changes-- changes which are certainly much easier to deal with when it's only a couple of changes vs what you might find over a much larger window... but of course one should always do what one is most comfortable with.... as I work in software, updating software (ie libraries, frameworks, etc) is what I do on a regular basis anyway ;)
